File: Data Structures & Algorithms/max-water-container/submission-2.py
class Solution:
    def maxArea(self, heights: List[int]) -> int:
        l = 0
        r = len(heights) - 1
        m = 0
        while l < r:
            lower = min(heights[l], heights[r])
            temp = (r-l) * lower
            m = max(m, temp)
            if lower == heights[l]:
                l+= 1
            else:
                r-=1
        return m
            

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-2.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int mp = 0;
        int n = prices[0];
        for (int i:prices) {
            mp = max(mp, i-n);

            n = min(i,n);
        }
        return mp;
    }
};
